### PR TITLE
Change the fields of src and dest to list of arguments.

### LINKED ISSUE #1354: Wrong blueprint and plotter for write flow
The following is from plotter generated in the notebook sample test: `source` should be `destination` (also please see attached see zip file with blueprint and plotter)

`- assetId: fybrik-notebook-sample/new-data-parquet
    flowType: write
    name: fybrik-notebook-sample/new-data-parquet-write
    subFlows:
    - flowType: write
      name: ""
      steps:
      - - cluster: thegreendragon
          name: ""
          parameters:
            api:
              connection:
                fybrik-arrow-flight:
                  hostname: my-notebook-fybrik-notebook-sample-arrow-flight-aef23.fybrik-blueprints
                  port: 80
                  scheme: grpc
                name: fybrik-arrow-flight
            source:
              assetId: fybrik-notebook-sample/new-data-parquet
          template: write
      triggers:
      - workload`

The following is from blueprint where vault credential should be under `write`:

`        - assetID: fybrik-notebook-sample/new-data-parquet
          capability: write
          source:
            connection:
              name: s3
              s3:
                bucket: bucket2
                endpoint: http://s3.fybrik-system:9090
                object_key: data.parquet
            format: parquet
            vault:
              read:
                address: http://vault.fybrik-system:8200
                authPath: /v1/auth/kubernetes/login
                role: module
                secretPath: /v1/kubernetes-secrets/data-creds?namespace=fybrik-notebook-sample
`

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**
Addresses issue #1366

Closes #1354 

Replace the source and destination fields with list of arguments in the blueprint and plotter. Also moving the changes done by @revit13 from PR #1357 to fix vault credentials generation which relates to issue #1354

The changes in this PR require adjustments in the modules as arrow-flight in order to pass the tests.

Signed-off-by: nassarmu10 <mohammad.nassar@ibm.com>

The new values format as sent to the arrow-flight server:
[manager_new_values.yaml.txt](https://github.com/fybrik/fybrik/files/8310946/manager_new_values.yaml.txt)

The new values format as sent to the implicit copy module:

### CODE REVIEW

File: manager/controllers/app/plotter_generator.go [hunk 1]
```diff
@@ -118,18 +118,21 @@ func (p *PlotterGenerator) GetCopyDestination(item *DataInfo, destinationInterfa
...
+func (p *PlotterGenerator) getAssetDataStore(item *DataInfo, flow taxonomy.DataFlow, vaultMap map[string]app.Vault) *app.DataStore {
+	if vaultMap == nil {
+		// create a new map in case it does not exist already
+		vaultMap = make(map[string]app.Vault)
+	}
 	if utils.IsVaultEnabled() {
 		// Set the value received from the catalog connector.
 		vaultSecretPath := item.DataDetails.Credentials
-		vaultMap[string(taxonomy.ReadFlow)] = app.Vault{
+		vaultMap[string(flow)] = app.Vault{
```
Comment (shlomitk1) [inline manager/controllers/app/plotter_generator.go:129]: How are these changes related to renaming source/dest?
These changes are problematic, for example, there are no vault credentials for Copy flow but only for Read and Write.

Comment (revit13) [reply manager/controllers/app/plotter_generator.go:129]: @shlomitk1 the changes are taken from #1357. we can leave it out from this pr for now. Thanks

File: manager/controllers/app/plotter_controller.go [hunk 2]
```diff
@@ -179,33 +179,40 @@ func (r *PlotterReconciler) convertPlotterModuleToBlueprintModule(plotter *app.P
 
 	var dataStore *app.DataStore
 	var destDataStore *app.DataStore
-	if plotterModule.ModuleArguments.Source != nil {
-		if plotterModule.ModuleArguments.Source.AssetID != "" {
-			assetID := plotterModule.ModuleArguments.Source.AssetID
+	if plotterModule.ModuleArguments.Arguments != nil && len(plotterModule.ModuleArguments.Arguments) > 0 && plotterModule.ModuleArguments.Arguments[0] != nil {
```
Comment (shlomitk1) [inline manager/controllers/app/plotter_controller.go:182]: if you check len(xxx) > 0, you don't need to check that xxx is not nil

File: manager/apis/app/v1alpha1/blueprint_types.go [hunk 3]
```diff
@@ -11,13 +11,9 @@ import (
 
 // AssetContext defines the input parameters for modules that access an asset
 type AssetContext struct {
-	// Source is the where the data currently resides
+	// Arguments for source and/or destination datastores
```
Comment (shlomitk1) [inline manager/apis/app/v1alpha1/blueprint_types.go:14]: Change to something like `datastores associated with the asset`

**Review by shlomitk1 — APPROVED** [review]

File: manager/controllers/app/plotter_controller_unit_test.go [hunk 4]
```diff
@@ -139,3 +139,117 @@ func TestPlotterController(t *testing.T) {
...
+			Namespace: namespace,
+		},
+	}
+	res, err := r.Reconcile(context.Background(), req)
+	g.Expect(err).To(gomega.BeNil())
+
+	// Check the result of reconciliation to make sure it has the desired state.
+	g.Expect(res.Requeue).To(gomega.BeFalse(), "reconcile did not requeue request as expected")
+
+	// Check if Job has been created and has the correct size.
```
Comment (shlomitk1) [inline manager/controllers/app/plotter_controller_unit_test.go:204]: Job -> Plotter, size is not relevant

**Review by shlomitk1 — APPROVED** [review]

File: manager/controllers/app/plotter_controller_unit_test.go (hunk 4 above)
Comment (revit13) [reply manager/controllers/app/plotter_controller_unit_test.go:204]: Fixed. Thanks

**Pull request merged**